src/core/snippets/snippets.ts

```diff
@@ -31,48 +44,67 @@ export class UIPSnippets extends UIPPlugin {
   }
 
   protected bindEvents() {
-    this.addEventListener('click', this._onClick);
+    this.querySelector(`.${UIPSnippets.LIST}`)?.addEventListener('click', this._onListClick);
+    this.querySelector(`.${UIPSnippets.DROPDOWN_CTRL}`)?.addEventListener('click', this._onDropdownClick);
   }
 
   protected unbindEvents() {
-    this.removeEventListener('click', this._onClick);
+    this.querySelector(`.${UIPSnippets.LIST}`)?.removeEventListener('click', this._onListClick);
+    this.querySelector(`.${UIPSnippets.DROPDOWN_CTRL}`)?.removeEventListener('click', this._onDropdownClick);
   }
```

Comment:
Consider store internal elements

```ts
class SomeBaseElement extends HTMLElement {

  @memoize
  protected get $internalElement() {
      const $el = document.createElement('div');
      $el.className = '.uip-smth'; // It might mot need to move it to the constant
      // ...
      return $el;
  }
  
  protected connectedCallback() {
     // First call will create internal instance
     this.appendChild(this.$internalElement);
     // Then use your internal ements for-free
     this.$internalElement.addEventListener(...);
  }
}
```

Reply:
Updated using considered solution.